File: Python/백준/Bronze/15474. 鉛筆 （Pencils）/鉛筆 （Pencils）.py
n,a,b,c,d = map(int,input().split())
if(n%a==0):
    p1 = (n//a)*b
else:
    p1 = (n//a+1)*b
if(n%c==0):
    p2 = (n//c)*d
else:
    p2 = (n//c+1)*d
if(p1<p2):
    print(p1)
else:
    print(p2)

File: C#/백준/Bronze/15474. 鉛筆 （Pencils）/鉛筆 （Pencils）.cs
using System;

namespace q{
    class w{
        static void Main(){
            int[] aa = Array.ConvertAll(Console.ReadLine().Split(),int.Parse);
            int c1 = aa[0]%aa[1]==0?aa[0]/aa[1]*aa[2]:(aa[0]/aa[1]+1)*aa[2];
            int c2 = aa[0]%aa[3]==0?aa[0]/aa[3]*aa[4]:(aa[0]/aa[3]+1)*aa[4];
            Console.WriteLine(c1<c2?c1:c2);
        }
    }
}